images/gcloud-bazel/BUILD.bazel

```diff
@@ -0,0 +1,2 @@
+# TODO(fejta): replace Dockerfile with an equivalent bazel rule. This seems promising:
+#   https://github.com/GoogleContainerTools/base-images-docker/blob/master/package_managers/bootstrap_image.bzl
```

Comment:
Yes I think (for TODO) package manager is the way to go. 

@xingao267 has done quite many works around package manager.

Reply:
Yes. Using package manager is promising. We've been using that to install java, wget, git, bazel, docker, and etc in our containers. But we haven't tried gcloud yet (it's on our list), and we are not sure if extra work will be needed to make that work (it could be just as simple as what we did below for Bazel).

I think we can use Dockerfile for now. I'll look into installing gcloud, and once that works. We can change this Dockerfile to bazel rules.

For your interest, we use rules here https://github.com/bazelbuild/bazel-toolchains/blob/master/container/rules/docker_toolchains.bzl to create containers. These rules wrap rules in https://github.com/GoogleContainerTools/base-images-docker/blob/master/package_managers. Example usage of the rules are:

https://github.com/bazelbuild/bazel-toolchains/blob/master/container/ubuntu16_04/builders/bazel/BUILD#L23
https://github.com/bazelbuild/bazel-toolchains/blob/master/container/ubuntu16_04/layers/bazel/BUILD

Reply:
Ack

I'll circle back later and try and figure out how to use these rules.